src/DynamoApplications/StartupUtils.cs

```diff
@@ -174,21 +179,85 @@ private static IUpdateManager InitializeUpdateManager()
             return um;
         }
 
+        /// <summary>
+        /// Extracts version of ASM dlls from a path by scanning for ASM dlls in the path.
+        /// Throws if ASM binaries cannot be found in the path.
+        /// </summary>
+        /// <param name="asmPath">path to directory containing asm dlls</param>
+        /// <returns></returns>
+        /// <param name="searchPattern">optional - to be used for testing</param>
+        /// <returns></returns>
+        internal static Version GetVersionFromASMPath(string asmPath, string searchPattern = "ASMAHL*.dll")
+        {
+            var ASMFilePath = Directory.GetFiles(asmPath, searchPattern, SearchOption.TopDirectoryOnly).FirstOrDefault();
+            if(ASMFilePath != null && File.Exists(ASMFilePath))
+            {
+                var asmVersion = FileVersionInfo.GetVersionInfo(ASMFilePath);
+                var libGversion = new Version(asmVersion.FileMajorPart, asmVersion.FileMinorPart, asmVersion.FileBuildPart);
+                return libGversion;
+            }
+            throw new FileNotFoundException("$Could not find geometry library binaries at : {geometryLibraryPath}");
+        }
+
+        /// <summary>
+        /// Use this overload to construct a DynamoModel when the location of ASM to use is known.
+        /// </summary>
+        /// <param name="CLImode">CLI mode starts the model in test mode and uses a seperate path resolver.</param>
+        /// <param name="asmPath">Path to directory containing geometry library binaries</param>
+        /// <returns></returns>
+        public static DynamoModel MakeModel(bool CLImode, string asmPath)
+        {
+            //get sandbox executing location - this is where libG will be located.
+            var exePath = Assembly.GetExecutingAssembly().Location;
+            var rootFolder = Path.GetDirectoryName(exePath);
+            //defaults - these will fail.
+            var preloaderLocation = "libg_0_0_0";
+            var geometryFactoryPath = Path.Combine(preloaderLocation, DynamoShapeManager.Utilities.GeometryFactoryAssembly);
+
+            try
+            {
+                if (!Directory.Exists(asmPath))
+                {
+                    throw new FileNotFoundException($"{nameof(asmPath)}:{asmPath}");
+                }
+                Version libGVersion = GetVersionFromASMPath(asmPath);
+
+                //get version of libG that matches the asm version that was supplied from geometryLibraryPath.
+                preloaderLocation = DynamoShapeManager.Utilities.GetLibGPreloaderLocation(libGVersion, rootFolder);
```

Comment:
If you read the function signature of `GetLibGPreloaderLocation`, the input should be the target ASM version. I dont think we need to do the extra step here to map a libG version and pass in?

Reply:
we're not doing any mapping, except for getting rid of the last part of the version (revision) and returning a Version object. I can call the variable ASM version if you think that is more clear. Sounds like the right name.

Reply:
Can you remind me the reason of removing revision?

Reply:
libG folders are named with major, minor, build only - revision is like the private part of a version number.